Sage/carbon
# fix(select): ensure re-opening the select resets scroll position the top

Updates the react-virtual dependency, which was directly responsible for the bug. The new version makes the virtualiser return 0 items whenever the select is closed - this change in behaviour resulted in some problems after the update, particularly with tests, which have been addressed. An accessibility error with long virtualised lists, where selecting one option and scrolling away from it so that the selected option is no longer in the DOM, resulting in an invalid aria-activedescendant attribute, has also been addressed.

fix #6399

### Proposed behaviour

Whenever opening a select, the top item in the list should be visible.

No critical accessibility errors should be flagged by Axe on any Select components.



### Current behaviour

When opening a select, scrolling to the bottom, closing and then reopening, the top item cannot be seen (but can be brought into view by scrolling up).

In some circumstances (see below codesandbox and instructions to reproduce) it is possible for the select input to have an `aria-activedescendant` attribute that doesn't correspond to anything in the DOM, causing Axe to flag a critical error.



### Checklist




#### QA


### Additional context



### Testing instructions


Test that both the linked Github issue, and the accessibility issue in the below sandbox, are fixed.

Perform regression testing on all 3 types of Select, as well as the Pager component which uses a Select internally.

The following CodeSandbox shows the accessibility error with virtualised lists https://s5nzqg.csb.app/

To reproduce:
- open the select and select "option 1"
- close the select, reopen it and scroll all the way to the bottom
- with the list still open, run Axe

You should see a critical issue "ARIA attributes must conform to valid values" (see screenshot - note that the other error is a known false positive) 
![image](https://github.com/Sage/carbon/assets/25186376/50b6deef-0432-4007-8042-4e25eb3549d5)
